fbg-server/src/rooms/rooms.service.ts

```diff
@@ -88,6 +88,36 @@ export class RoomsService {
     });
   }
 
+  /** Forcefully removes user from room. */
+  async removeFromRoom(userIdOfCaller: number, userIdToBeRemoved: number, roomId: string): Promise<RoomEntity> {
+    return await inTransaction(this.connection, async (queryRunner) => {
+      const room = await this.getRoomEntity(roomId);
+      if (room.match) {
+        return room;
+      }
+
+      const userMembership = room.userMemberships.find(
+        (membership) => membership.user.id === userIdOfCaller,
+      );
+
+      if (!userMembership || !userMembership.isCreator) {
+        throw new HttpException(
+          'You must be the creator of the room',
+          HttpStatus.BAD_REQUEST,
+        );
+      }
+
+      await this.removeMembership(queryRunner, userIdToBeRemoved, room);
+
+      if (room.isPublic) {
+        await this.lobbyService.notifyLobbyUpdate();
```

Comment:
why are you not calling leaveRoom instead? it already has this logic of notifying...

Reply:
I see now, because of the transaction...